### PR TITLE
docs: update dependency graph

### PR DESCRIPTION
**Pull request by philknows**
**Motivation**

In order to assist the upcoming EF Attackathon, it would be good to update dependencies for contributors of all sorts to understand the high-level overview of Lodestar, its packages and dependencies for a better experience. The last update to this was over 4 years ago.

**Description**

This PR modifies the `depgraph.md` page to be up-to-date on the current architecture of Lodestar and how everything fits together. Newer packages such as the Prover have not even been mentioned due to its recent development.

### CODE REVIEW

**Comment by nflaig** [comment]
I updated the mermaid diagram, it's pretty crowded but still readable, bumped the font-size up quite a bit

image

File: README.md [hunk 1]
```diff
@@ -44,21 +44,25 @@ yarn build
...
+| [`@lodestar/prover`](./packages/prover)                     | [![npm](https://img.shields.io/npm/v/@lodestar/prover)](https://www.npmjs.com/package/@lodestar/prover)                     | [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)  | [![documentation](https://img.shields.io/badge/readme-blue)](./packages/prover)           | :white_check_mark: Eth Consensus light client verifier for execution JSON-RPC calls |
+| [`@lodestar/reqresp`](./packages/reqresp)                   | [![npm](https://img.shields.io/npm/v/@lodestar/reqresp)](https://www.npmjs.com/package/@lodestar/reqresp)                   | [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)  | [![documentation](https://img.shields.io/badge/readme-blue)](./packages/reqresp)          | :telephone_receiver: TS implementation of Eth Consensus Req/Resp protocol           |
```
Comment (nflaig) [inline README.md]: To align with others
```suggestion
| [`@lodestar/reqresp`](./packages/reqresp)                   | [![npm](https://img.shields.io/npm/v/@lodestar/reqresp)](https://www.npmjs.com/package/@lodestar/reqresp)                   | [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)  | [![documentation](https://img.shields.io/badge/readme-blue)](./packages/reqresp)          | :telephone_receiver: Eth Consensus Req/Resp protocol           |
```

**Review by nflaig — DISMISSED** [review]
LGTM - would be good to get another review on this

File: README.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ yarn build
...
-| [`@lodestar/prover`](./packages/prover)                     | [![npm](https://img.shields.io/npm/v/@lodestar/prover)](https://www.npmjs.com/package/@lodestar/prover)                     | [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)  | [![documentation](https://img.shields.io/badge/readme-blue)](./packages/prover)           | :white_check_mark: Eth Consensus light client verifier for execution JSON-RPC calls |
+| [`@lodestar/prover`](./packages/prover)                     | [![npm](https://img.shields.io/npm/v/@lodestar/prover)](https://www.npmjs.com/package/@lodestar/prover)                     | [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)  | [![documentation](https://img.shields.io/badge/readme-blue)](./packages/prover)           | :white_check_mark: Eth Consensus Light client verifier for execution JSON-RPC calls |
```
Comment (nflaig) [inline README.md]: it's still not aligned, the one says "Ethereum Light client" and the other one "Eth Consensus Light client", it's probably redudant to add "Consensus" here as pure Execution light client are not a thing anymore

Comment (philknows) [reply README.md]: Sticking to "Ethereum light client" throughout.

Comment (nflaig) [reply README.md]: I'm fine with either as long as we are consistent

**Review by nflaig — DISMISSED** [review]

**Review by nflaig — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.21.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.21.0) :tada:

**Pull request merged**